File: algorithms/sorting/selection-sort/SelectionSort.dart
class Sorting {
  static List<int> SelectionSort(List<int> unsortedList) {
    int n = unsortedList.length;
    int min, temp;

    for (int i = 0; i < n; i++) {
      min = i;

      for (int j = i; j < n; j++) {
        if (unsortedList[j] < unsortedList[min]) {
          min = j;
        }
      }

      temp = unsortedList[min];
      unsortedList[min] = unsortedList[i];
      unsortedList[i] = temp;
    }

    return unsortedList;
  }
}

File: algorithms/sorting/selection-sort/SelectionSort.ts
class Sorting {
  public static SelectionSort(unsortedArray: number[]): number[] {
    let n: number = unsortedArray.length;
    let min: number, temp: number;

    for (let i: number = 0; i < n; i++) {
      min = i;

      for (let j: number = i; j < n; j++) {
        if (unsortedArray[j] < unsortedArray[min]) {
          min = j;
        }
      }

      temp = unsortedArray[min];
      unsortedArray[min] = unsortedArray[i];
      unsortedArray[i] = temp;
    }

    return unsortedArray;
  }
}
